Sources/Core/Moya/AuthorizationMoyaPlugin.swift

```diff
@@ -10,22 +10,33 @@ import Foundation
 import Moya
 
 final class AuthorizationMoyaPlugin: PluginType {
-    var token: Token
-    
+    private let serialQueue = DispatchQueue(label: "com.getstream.io.AuthorizationMoyaPlugin")
+    private var token: Token
+
     init(_ token: Token = "") {
         self.token = token
     }
-    
+
     func prepare(_ request: URLRequest, target: TargetType) -> URLRequest {
-        let token = self.token
+        var token = ""
+        serialQueue.sync {
```

Comment:
You could use the return of the sync method like:

```
let token = serialQueue.sync { self.token }
```

Reply:
@angelolloqui Thanks for the suggestion 👍